drake/systems/plants/shapes/HeightMapTerrain.h

```diff
@@ -0,0 +1,139 @@
+#ifndef __HeightMapTerrain_H__
+#define __HeightMapTerrain_H__
+
+#include <string>
+
+#include <Eigen/Dense>
+#include <Eigen/StdVector>
+#include <cstring>  //for the definition of syze_t
+
+#include "drake/drakeShapes_export.h"
+
+#include "drake/systems/plants/shapes/Geometry.h"
+
+namespace DrakeShapes {
+
+//!  A class representing a height map terrrain shape
+/*!
+*/
+class DRAKESHAPES_EXPORT HeightMapTerrain : public Geometry {
+ public:
+  HeightMapTerrain() {}
+
+  //! HeightMapTerrain constructor.
+  /*!
+    A more elaborate description of the constructor.
+
+    \param size The provided array will be scaled so that the terrain size(0) x
+    size(1) in size and has maximum height size(2). The minimum height will be
+    located at zero height.
+    \param pos Position of the terrain center in world's coordinates. The
+    terrain center is located at the middle of the 2D plane and at zero height.
+  */
+  HeightMapTerrain(const std::string &name, const Eigen::Vector2i &ncells,
+                   const Eigen::Vector2d &size);
+  HeightMapTerrain(const HeightMapTerrain &other);
+  virtual ~HeightMapTerrain();
+  virtual HeightMapTerrain *clone() const;
+  virtual void getPoints(Eigen::Matrix3Xd &points) const;
+  virtual void getBoundingBoxPoints(Eigen::Matrix3Xd &points) const;
+  virtual void getTerrainContactPoints(Eigen::Matrix3Xd &points) const;
+
+  virtual void initialize();
+  int nTotCells() const { return ncells.prod(); }
+  double cellValue(int i) const;
+  double cellValue(int i, int j) const;
+  double heightValue(int i, int j) const;
+  double &cellValue(int i, int j);
+  bool writeToFile(const std::string &fname) const;
+
+  enum RawDataType { UCHAR, SHORT, FLOAT };
+  typedef unsigned char byte_t;
+
+  std::size_t nBytes;  //!< number of bytes allocated in m_rawHeightfieldData
+  long nTotNodes;      //!< total number of nodes
```

Comment:
`long` is not portable (size varies by compiler) - don't use it. If you have a particular number of bits in mind you can use `int64_t`, or use `long long` if you just want something very big (that's 64 bits on all the compilers I know of).